### PR TITLE
[luajit] Add 2.1.0-beta3

### LINKED ISSUE #11884: [package] luajit/2.0.5: Recipe error on MacOS (not enough values to unpack)
### Package and Environment Details

* Package Name/Version: **luajit/2.0.5**
* Operating System+version: **macOS 11**
* Compiler+version: **apple-clang 13**
* Conan version: **conan 1.50.0**
* Python version: **Python 3.8.13**

### Conan profile

[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=13
os=Macos
os_build=Macos
[options]
[build_requires]
[env]

### Steps to reproduce

The issue occured on github actions, `runs-on: macos-latest` (https://github.com/Bobini1/RhythmGame/runs/7483865663?check_suite_focus=true)

- conanfile.py (requirements):
self.requires("luajit/2.0.5")
```
pip3 install "conan<2,!=1.49.0"
conan profile new default --detect --force
conan install . -b missing
```

### Logs

<details><summary>Click to expand log</summary>

```
luajit/2.0.5: Configuring sources in /Users/runner/.conan/data/luajit/2.0.5/_/_/source
luajit/2.0.5: Copying sources to build folder
luajit/2.0.5: Building your package in /Users/runner/.conan/data/luajit/2.0.5/_/_/build/299aaabcb5fb1385c096ccf2c550a653c4438[173](https://github.com/Bobini1/RhythmGame/runs/7483865663?check_suite_focus=true#step:7:174)
luajit/2.0.5: Generator txt created conanbuildinfo.txt
luajit/2.0.5: Aggregating env generators
luajit/2.0.5: Calling build()
luajit/2.0.5: 
WARN: Remotes registry file missing, creating default one in /Users/runner/.conan/remotes.json
luajit/2.0.5: ERROR: Package '299aaabcb5fb1385c096ccf2c550a653c4438173' build failed
luajit/2.0.5: WARN: Build folder /Users/runner/.conan/data/luajit/2.0.5/_/_/build/299aaabcb5fb1385c096ccf2c550a653c4438173
ERROR: luajit/2.0.5: Error in build() method, line 76
	major, minor, _ = platform.mac_ver()[0].split(".")
	ValueError: not enough values to unpack (expected 3, got 2)
Error: Process completed with exit code 1.
```

</details>

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **luajit/2.1.0-beta3**

- luajit 2.0.5 is not prepared and does not support Mac M1
- Mac M1 support has been introduced on 2.1
- No stable release will be released soon, so beta is the only alternative
- Backporting to 2.0.5 M1 support requires a lot of effort
- Fix Mac version parsing

fixes #11884

/cc @Bobini1

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/luajit/all/conanfile.py [hunk 1]
```diff
@@ -73,7 +73,10 @@ def build(self):
                 # is not set then it's forced to 10.4, which breaks compile on Mojave.
                 version = self.settings.get_safe("os.version")
                 if not version and platform.system() == "Darwin":
-                    major, minor, _ = platform.mac_ver()[0].split(".")
+                    platform.mac_ver
```
Comment (uilianries) [inline recipes/luajit/all/conanfile.py:76]:
```suggestion
```

**Review by Bobini1 — APPROVED** [review]
Looks alright, I made an equivalent change locally and it worked fine. I can't confirm whether it will work on all mac versions but I think it's a reasonable fix.

**Comment by Leohige** [comment]
Same issue here, the error seems to have been fixed.

**Review by jgsogo — APPROVED** [review]

**Review by Bobini1 — APPROVED** [review]

**Comment by Bobini1** [comment]
: /

**Pull request closed without merging**